File: go-lambda/nthPrime.go
package main

import (
	"fmt"
	"github.com/aws/aws-lambda-go/lambda"
	"os"
	"strconv"
	"time"
)

type RequestClass struct {
	InputInt    int    `json:"inputInt"`
	InputString string `json:"inputString"`
}

type ResponseClass struct {
	OutputInt    int     `json:"outputInt"`
	Time1        float64 `json:"time1"`
	Time2        float64 `json:"time2"`
	LongTime1    int64   `json:"longTime1"`
	LongTime2    int64   `json:"longTime2"`
	OutputDouble float64 `json:"outputDouble"`
	OutputString string  `json:"outputString"`
}

func isPrime(num int) bool {
	factor := 2
	for (factor * factor) <= num {
	      if num % factor == 0{
	      	 return false
	      }
	      factor++
	}
	return true
}

func nthPrime(n int) int {
	if n == 1 {
		return 2
	}
	count := 1
	num := 3
	for count < n {
		if isPrime(num) {
			count += 1
		}
		num += 2
	}
	return num - 2
}

func Handler(input RequestClass) (ResponseClass, error) {
	response := ResponseClass{}
	start := time.Now()
	response.OutputInt = nthPrime(input.InputInt)
	end := time.Now()
	response.LongTime1 = int64(end.Sub(start) / time.Millisecond)
	response.OutputString = fmt.Sprintf("(go) calculated %d prime", input.InputInt)

	return response, nil
}

func main() {
	if len(os.Args) == 3 {
		var req RequestClass
		input, err := strconv.Atoi(os.Args[1])
		req.InputInt = input
		req.InputString = os.Args[2]
		output, err := Handler(req)
		err = err
		fmt.Println(output)
	} else {
		lambda.Start(Handler)
	}
}

File: java-lambda/src/main/java/jalexander/ninja/NthPrime.java
package jalexander.ninja;

import com.amazonaws.services.lambda.runtime.Context;

public class NthPrime {
    public static ResponseClass handler(RequestClass request, Context context) {
        ResponseClass response = new ResponseClass();
        response.outputString = "(java) calculated " + request.inputInt + "th prime";

        long start = System.currentTimeMillis();
        response.outputInt = nthPrime(request.inputInt);
        long end = System.currentTimeMillis();
        response.longTime1 = end - start;

        return response;
    }

    public static int nthPrime(int n) {
        if (n == 1) {
            return 2;
        }
        int count = 1;
        int num = 3;
        while (count < n) {
            if (isPrime(num)) {
                count += 1;
            }
            num += 2;
        }
        return num - 2;
    }

    public static boolean isPrime(int num) {
        int factor = 2;
        while (factor * factor <= num) {
            if (num % factor == 0) {
                return false;
            }
            factor++;
        }
        return true;
    }

    public static void main(String[] args){
        RequestClass request = new RequestClass();
        request.inputInt = Integer.parseInt(args[0]);
        request.inputString = args[1];
        ResponseClass response = handler(request, null);
        System.out.println(response.csvString());
    }
}
